src/store/index.js

```diff
@@ -9,17 +9,62 @@ import type { Data, Message } from './types';
 
 const store: Data = require('./emails.json');
 
-/**
- * You might find it useful to implement some functions that make accessing
- * deeply nested attributes of Messages or Threads...
- */
+function getNewMessageIds() {
+  return store.mailboxes.INBOX.threadIds;
+}
+
+function getNewMessages(newMessageIds) {
+  let newMessages = [];
+  for (let id = 0; id < newMessageIds.length; id += 1) {
+    const rawMessageObj = store.messages[newMessageIds[id]];
+    const refinedMessageObj = {
+      Snippet: rawMessageObj.snippet,
+      Date: '',
+      From: '',
+      To: '',
+      Subject: '',
+    };
+    for (let header of rawMessageObj.payload.headers) {
+      refinedMessageObj[header.name] = header.value;
+    };
+    newMessages.push(refinedMessageObj);
+  }
+  return newMessages;
+}
+
+function formatNewMessage(newMessage) {
+  const formattedRecipients = newMessage.To;
+  const formattedDate = formatTimestamp(newMessage.Date);
+  const formattedSender = newMessage.From.match(/([A-Z])\w+/g).join(' ');
+  const formattedSubject = formatSubject(newMessage.Subject);
+  const formattedSnippet = formatSnippet(newMessage.Snippet);
+  const formattedMessage = {
+    recipients: formattedRecipients,
+    sender: formattedSender,
+    subject: formattedSubject,
+    snippet: formattedSnippet,
+    date: formattedDate
+  }
+  return formattedMessage;
+}
+
+function formatTimestamp(datestring) {
```

Comment:
Consider using a library for this; it's true that adding more dependencies generally more things to worry about, but I think it's ok here (see [date-fns](https://date-fns.org/), [Moment.js](https://momentjs.com/) for a couple possible choices).

Reply:
Additionally, check out [`Date.prototype.toLocaleString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString)